agent/config/builder.go

```diff
@@ -1132,6 +1132,43 @@ func (b *Builder) expandAddrs(name string, s *string) []net.Addr {
 	return addrs
 }
 
+// expandOptionalAddrs expands the go-sockaddr template in s and returns the
+// result as a list of strings. If s does not contain a go-sockaddr template,
+// the result list will contain the input string as a single element with no
+// error set. In contrast to expandAddrs, expandOptionalAddrs does not validate
+// if the result contains valid addresses and returns a list of strings.
+// However, if the expansion of the go-sockaddr template fails an error is set.
```

Comment:
Can you explain why you don't want to validate? for example if the resolved template returns a unix socket address we should throw an error because that's not going to work for retry-join/retry-join-wan.

Reply:
This was from @dominik-lekse's original PR but I think the rationale is that in this case the `runtime.Config` members we are outputting to are `[]string` not `[]net.Addr`.

We _could_ add the parsing here just to catch errors but on the basis that before this PR passing an invalid IP to these config params is not validated here in the builder it seems reasonable not to jump through that extra hoop.

Presumably the agent itself will error if the expanded (or just non-template) value provided was invalid just like it does before this PR.

Reply:
Did a quick test on master and looks like an invalid address in retry join gets all the way upto the join call in memberlist  before it errors out. We can do better with this validation in the future though, agree with you on the rationale to leave it as is for this PR.